diff --git a/CMakeLists.txt b/CMakeLists.txt
index c2747d9..3417f79 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,9 +51,6 @@ endif()
 add_definitions(
 	-DFNA3D_DRIVER_OPENGL
 )
-if(BUILD_SDL3)
-	add_definitions(-DFNA3D_DRIVER_SDL)
-endif()
 add_definitions(
 	-DMOJOSHADER_NO_VERSION_INCLUDE
 	-DMOJOSHADER_USE_SDL_STDLIB
@@ -113,7 +110,6 @@ add_library(FNA3D
 	src/FNA3D.c
 	src/FNA3D_Driver_D3D11.c
 	src/FNA3D_Driver_OpenGL.c
-	src/FNA3D_Driver_SDL.c
 	src/FNA3D_Image.c
 	src/FNA3D_PipelineCache.c
 	src/FNA3D_Tracing.c
@@ -124,7 +120,6 @@ add_library(mojoshader STATIC
 	MojoShader/mojoshader_common.c
 	MojoShader/mojoshader_d3d11.c
 	MojoShader/mojoshader_opengl.c
-	MojoShader/mojoshader_sdlgpu.c
 	MojoShader/profiles/mojoshader_profile_common.c
 	MojoShader/profiles/mojoshader_profile_glsl.c
 	MojoShader/profiles/mojoshader_profile_hlsl.c
@@ -138,17 +133,11 @@ if(TRACING_SUPPORT)
 		$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/MojoShader>
 	)
 endif()
-if(BUILD_SDL3)
-	add_executable(fna3d_dumpspirv dumpspirv/dumpspirv.c)
-	target_link_libraries(fna3d_dumpspirv FNA3D)
-	target_include_directories(fna3d_dumpspirv PUBLIC
-		$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/MojoShader>
-	)
-endif()
 
 # Build flags
 if(NOT MSVC)
-	set_property(TARGET FNA3D PROPERTY COMPILE_FLAGS "-std=gnu99 -Wall -Wno-strict-aliasing -pedantic")
+	set_property(TARGET FNA3D PROPERTY COMPILE_FLAGS "-pthread -std=gnu99 -Wall -Wno-strict-aliasing -pedantic")
+	set_property(TARGET mojoshader PROPERTY COMPILE_FLAGS "-pthread")
 endif()
 if(BUILD_SHARED_LIBS)
 	set_property(TARGET mojoshader PROPERTY POSITION_INDEPENDENT_CODE ON)
diff --git a/src/FNA3D.c b/src/FNA3D.c
index 25eb848..5a27ee4 100644
--- a/src/FNA3D.c
+++ b/src/FNA3D.c
@@ -40,9 +40,6 @@
 /* Drivers */
 
 static const FNA3D_Driver *drivers[] = {
-#if FNA3D_DRIVER_SDL
-	&SDLGPUDriver,
-#endif
 #if FNA3D_DRIVER_D3D11
 	&D3D11Driver,
 #endif
