diff --git a/src/audio/emscripten/SDL_emscriptenaudio.c b/src/audio/emscripten/SDL_emscriptenaudio.c
index 84ba63b..fa91e62 100644
--- a/src/audio/emscripten/SDL_emscriptenaudio.c
+++ b/src/audio/emscripten/SDL_emscriptenaudio.c
@@ -189,7 +189,7 @@ static bool EMSCRIPTENAUDIO_OpenDevice(SDL_AudioDevice *device)
     }
 
     // limit to native freq
-    device->spec.freq = EM_ASM_INT({ return Module['SDL3'].audioContext.sampleRate; });
+    device->spec.freq = MAIN_THREAD_EM_ASM_INT({ return Module['SDL3'].audioContext.sampleRate; });
     device->sample_frames = SDL_GetDefaultSampleFramesFromFreq(device->spec.freq) * 2;  // double the buffer size, some browsers need more, and we'll just have to live with the latency.
 
     SDL_UpdatedAudioDeviceFormat(device);
diff --git a/src/locale/emscripten/SDL_syslocale.c b/src/locale/emscripten/SDL_syslocale.c
index 6983628..560e6c3 100644
--- a/src/locale/emscripten/SDL_syslocale.c
+++ b/src/locale/emscripten/SDL_syslocale.c
@@ -27,7 +27,7 @@
 bool SDL_SYS_GetPreferredLocales(char *buf, size_t buflen)
 {
     /* *INDENT-OFF* */ // clang-format off
-    EM_ASM({
+    MAIN_THREAD_EM_ASM({
         var buf = $0;
         var buflen = $1;
         var list = undefined;
diff --git a/src/misc/emscripten/SDL_sysurl.c b/src/misc/emscripten/SDL_sysurl.c
index 881de01..0e6fb79 100644
--- a/src/misc/emscripten/SDL_sysurl.c
+++ b/src/misc/emscripten/SDL_sysurl.c
@@ -28,6 +28,6 @@ EM_JS_DEPS(sdlsysurl, "$UTF8ToString");
 
 bool SDL_SYS_OpenURL(const char *url)
 {
-    EM_ASM(window.open(UTF8ToString($0), "_blank"), url);
+    MAIN_THREAD_EM_ASM(window.open(UTF8ToString($0), "_blank"), url);
     return true;
 }
diff --git a/src/video/SDL_video.c b/src/video/SDL_video.c
index e773969..ea50ee0 100644
--- a/src/video/SDL_video.c
+++ b/src/video/SDL_video.c
@@ -5693,7 +5693,7 @@ bool SDL_ShowSimpleMessageBox(SDL_MessageBoxFlags flags, const char *title, cons
     if (!message) {
         message = "";
     }
-    EM_ASM({
+    MAIN_THREAD_EM_ASM({
         alert(UTF8ToString($0) + "\n\n" + UTF8ToString($1));
     },
             title, message);
diff --git a/src/video/emscripten/SDL_emscriptenvideo.c b/src/video/emscripten/SDL_emscriptenvideo.c
index 413d96f..e159526 100644
--- a/src/video/emscripten/SDL_emscriptenvideo.c
+++ b/src/video/emscripten/SDL_emscriptenvideo.c
@@ -64,7 +64,7 @@ static SDL_SystemTheme Emscripten_GetSystemTheme(void)
     /* Technically, light theme can mean explicit light theme or no preference.
        https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme#syntax */
 
-    int theme_code = EM_ASM_INT({
+    int theme_code = MAIN_THREAD_EM_ASM_INT({
         if (!window.matchMedia) {
             return -1;
         }
